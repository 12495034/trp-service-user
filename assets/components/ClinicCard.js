import * as React from 'react';
import { Card, Text, Badge } from 'react-native-paper';
import { StyleSheet, View, Pressable } from 'react-native';
import { createDateString } from '../commonFunctions/createDateString';

function ClinicCard(props) {
    return (
        <Pressable
            style={({ pressed }) => [
                { backgroundColor: pressed ? '#F7C3E9' : '#0000', borderRadius: 5 },
            ]}
            //only allow clinics to be selected if they have appointment slots available
            onPress={() => props.slots == 0 ? null : props.details(props.id)}
        >
            <Card mode='outlined' style={styles.card}>
                <Card.Title title={props.location} titleStyle={{fontSize:18}} subtitleStyle={{fontSize:14}} subtitle={`${props.center}, ${props.addDetails}` } />
                <Card.Content>
                    <Text variant="titleMedium">{createDateString(props.date)}</Text>
                    <Text variant="titleLarge">Start Time: {props.time}</Text>
                    <View style={styles.availabilityRow}>
                        <Text variant="titleLarge">Capacity: {props.capacity}</Text>
                        <Badge
                            size={25}
                            visible={true}
                            style={props.slots == 0 ? { backgroundColor: 'red', } : { backgroundColor: 'green' }}
                        >{props.slots == 0 ? "Full" : props.slots}</Badge>
                    </View>
                </Card.Content>
                <Card.Actions>
                </Card.Actions>
            </Card>
        </Pressable>
    );
}

const styles = StyleSheet.create({
    card: {
        marginBottom: 2,
        // borderColor: 'black',
        // borderStyle: 'solid'
    },
    availabilityRow: {
        flexDirection: 'row',
        justifyContent: 'space-between'
    }
})

export default ClinicCard;