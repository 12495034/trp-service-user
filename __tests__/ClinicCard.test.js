import React from 'react';
import renderer from 'react-test-renderer';
import ClinicCard from '../assets/components/ClinicCard';

describe('Clinic Card Tests', () => {
  test('renders clinicCard component correctly', () => {
    const tree = renderer.create(<ClinicCard
      id={123456}
      status={'Active'}
      location={'Belfast'}
      center={'LGBT Center'}
      capacity={8}
      date={'3000-01-01'}
      time={'18:00'}
      slots={8}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  })
});