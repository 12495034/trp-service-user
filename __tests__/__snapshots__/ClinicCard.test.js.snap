// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders clinicCard component correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "backgroundColor": "#0000",
        "borderRadius": 5,
      },
    ]
  }
>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "#ffffff",
        "borderColor": "rgba(0, 0, 0, 0.12)",
        "borderRadius": 4,
        "borderWidth": 1,
        "elevation": 0,
        "marginBottom": 2,
      }
    }
  >
    <View
      accessibilityState={
        {
          "disabled": true,
        }
      }
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "flexGrow": 1,
          "flexShrink": 1,
        }
      }
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "paddingLeft": 16,
            },
            {
              "minHeight": 72,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            [
              {
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            numberOfLines={1}
            style={
              [
                {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                {
                  "textAlign": "left",
                },
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(0, 0, 0, 0.87)",
                    "fontFamily": "System",
                    "fontWeight": "500",
                    "writingDirection": "ltr",
                  },
                  [
                    {
                      "fontSize": 20,
                      "letterSpacing": 0.15,
                      "lineHeight": 30,
                      "marginVertical": 2,
                    },
                    [
                      {
                        "minHeight": 30,
                        "paddingRight": 16,
                      },
                      {
                        "marginBottom": 0,
                      },
                      undefined,
                    ],
                  ],
                ],
              ]
            }
          >
            Belfast
          </Text>
          <Text
            numberOfLines={1}
            style={
              [
                {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                {
                  "textAlign": "left",
                },
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(0, 0, 0, 0.54)",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "writingDirection": "ltr",
                  },
                  [
                    {
                      "fontSize": 12,
                      "letterSpacing": 0.4,
                      "lineHeight": 20,
                      "marginVertical": 2,
                    },
                    [
                      {
                        "marginVertical": 0,
                        "minHeight": 20,
                        "paddingRight": 16,
                      },
                      undefined,
                    ],
                  ],
                ],
              ]
            }
          >
            LGBT Center
          </Text>
        </View>
        <View />
      </View>
      <View
        style={
          [
            {
              "paddingHorizontal": 16,
            },
            undefined,
            undefined,
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              {
                "textAlign": "left",
              },
              undefined,
            ]
          }
          variant="titleMedium"
        >
          3000-01-01
        </Text>
        <Text
          style={
            [
              {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              {
                "textAlign": "left",
              },
              undefined,
            ]
          }
          variant="titleLarge"
        >
          Start Time: 
          18:00
        </Text>
        <View
          style={
            {
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              [
                {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                {
                  "textAlign": "left",
                },
                undefined,
              ]
            }
            variant="titleLarge"
          >
            Capacity: 
            8
          </Text>
          <Text
            collapsable={false}
            numberOfLines={1}
            style={
              {
                "alignSelf": "flex-end",
                "backgroundColor": "green",
                "borderRadius": 12.5,
                "color": "#ffffff",
                "fontFamily": "System",
                "fontSize": 12.5,
                "fontWeight": "400",
                "height": 25,
                "lineHeight": 12.5,
                "minWidth": 25,
                "opacity": 1,
                "overflow": "hidden",
                "paddingHorizontal": 4,
                "textAlign": "center",
                "textAlignVertical": "center",
              }
            }
          >
            8
          </Text>
        </View>
      </View>
      <View
        index={2}
        siblings={
          [
            "withTheme(Card.Title)",
            "Card.Content",
            "Card.Actions",
          ]
        }
        style={
          [
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "padding": 8,
            },
            undefined,
          ]
        }
        total={3}
      />
    </View>
  </View>
</View>
`;
